rules-reviewed/eap7/eap6/hsearch.windup.xml

```diff
@@ -0,0 +1,1153 @@
+            <when>
+                <javaclass references="org.hibernate.search.filter.FilterKey" />
+            </when>
+            <perform>
+                <hint title="Deprecated interface org.hibernate.search.filter.FilterKey in Hibernate Search 5" effort="1" severity="optional">
+                    <message>Custom filter keys are deprecated and are scheduled for removal in Hibernate Search 6. As of Hibernate Search 5.1, keys for caching Lucene filters are calculated automatically based on the given filter parameters.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_hibernate_search_changes"
+                        title="Changes in Hibernate Search 5.x" />
+                    <link href="http://hibernate.org/search/documentation/migrate/5.1/#api-changes" title="Hibernate Search 5.1 API changes" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+
+        <rule id="hsearch-00106">
+            <when>
+                <javaclass references="org.hibernate.search.filter.StandardFilterKey" />
+            </when>
+            <perform>
+                <hint title="Deprecated class org.hibernate.search.filter.StandardFilterKey in Hibernate Search 5" effort="1" severity="optional">
+                    <message>Custom filter keys `StandardFilterKey` are deprecated and are scheduled for removal in Hibernate Search 6.
+                    As of Hibernate Search 5.1, keys for caching Lucene filters are calculated automatically based on the given filter parameters.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_hibernate_search_changes"
+                        title="Changes in Hibernate Search 5.x" />
+                    <link href="http://hibernate.org/search/documentation/migrate/5.1/#api-changes" title="Hibernate Search 5.1 API changes" />
+                    <link
+                        href="https://docs.jboss.org/hibernate/search/5.5/api/index.html?org/hibernate/search/filter/StandardFilterKey.html"
+                        title="hibernate Search javadoc" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00107">
+            <when>
+                <javaclass references="org.hibernate.search.annotations.FieldCacheType" />
+            </when>
+            <perform>
+                <hint title="Deprecated annotation FieldCacheType in Hibernate Search 5" effort="1" severity="optional">
+                    <message>Remove the `@FieldCacheType` annotation. No alternative replacement necessary as the Field Cache in Apache Lucene feature does no longer exist.</message>
```

Comment:
`FieldCacheType` is not an annotation, it's an enum that is used as an attribute of the `@CacheFromIndex` annotation.